sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -40,23 +40,42 @@
     private SetStatementAnalyzer() {}
 
     public static SetAnalyzedStatement analyze(SetStatement node, ParameterContext parameterContext) {
-        if (SetStatement.Scope.SESSION.equals(node.scope())) {
-            logger.debug("SET SESSION STATEMENT: {}", node.toString());
+
+        if (SetStatement.Scope.GLOBAL.equals(node.scope())) {
+            for (Assignment assignment : node.assignments()){
+                if (assignment.expression().equals(new QualifiedNameReference(new QualifiedName("default")))){
+                    throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                        "DEFAULT cannot be used with SET GLOBAL. To reset a setting use RESET."));
+                }
+            }
+
+        } else {
+            for (Assignment assignment : node.assignments()){
+                if (assignment.expressions().contains(new QualifiedNameReference(new QualifiedName("default"))) && assignment.expressions().size() > 1){
+                    throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                        "DEFAULT cannot be used in a list assignment"));
+                }
+            }
+
+
+            logger.debug("SET {} STATEMENT: {}",node.scope().toString() , node.toString());
```

Comment:
there is also already a `SetSessionPlan` (which results in a `NoopTask` tho)
guess `TransportExecutor.visitSetPlan()` is the better place to log the warning